include/particles/particle.tcc

```diff
@@ -243,11 +243,14 @@ void mpm::Particle<Tdim>::initialise() {
   volume_ = std::numeric_limits<double>::max();
   volumetric_strain_centroid_ = 0.;
 
-  // Initialize vector data properties
-  this->properties_["stresses"] = [&]() { return stress(); };
-  this->properties_["strains"] = [&]() { return strain(); };
-  this->properties_["velocities"] = [&]() { return velocity(); };
-  this->properties_["displacements"] = [&]() { return displacement(); };
+  // Initialize scalar and tensor data properties
+  this->scalar_properties_["mass"] = [&]() { return mass(); };
+  this->scalar_properties_["volume"] = [&]() { return volume(); };
+  this->scalar_properties_["mass_density"] = [&]() { return mass_density(); };
+  this->tensor_properties_["stresses"] = [&]() { return stress(); };
+  this->tensor_properties_["strains"] = [&]() { return strain(); };
+  this->tensor_properties_["velocities"] = [&]() { return velocity(); };
+  this->tensor_properties_["displacements"] = [&]() { return displacement(); };
```

Comment:
Does this affect vtk vector?

Reply:
why do we have vector variables in MPM but not here?

Reply:
> why do we have vector variables in MPM but not here?

Initially, they are generally called as `properties_`, you call it through the tensor_data which returns an `Eigen::VectorXd`. When passing to the io, we are calling the `particles_tensor_data()` function in mesh class with templatized size, i.e. 3 for vector and 6 for tensor. See this: https://github.com/cb-geo/mpm/blob/a1023baa55cf387275a274aed6b9b955f432d2ad/include/solvers/mpm_base.tcc#L572-L575
and
https://github.com/cb-geo/mpm/blob/a1023baa55cf387275a274aed6b9b955f432d2ad/include/solvers/mpm_base.tcc#L593-L596
I am currently following your previous implementation. If you want to distinguish them to `vector_variables_`, we need to change the `particles_tensor_data()` function in `mesh.tcc` and create `vector_data()` function in particle. I am fine with staying as it is or changing it to `vector_data`, both will work.

> Does this affect vtk vector?

It does not. As I mentioned above, the template size allocates the correct size of Eigen matrix for the io.